controller/settingscontroller.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace OCA\Notes\Controller;
+use OCP\AppFramework\Controller;
+
+use OCP\IConfig;
+use OCP\IRequest;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use OCP\Files\IRootFolder;
+
+class SettingsController extends Controller
+{
+	private $config;
+	private $userSession;
+	private $root;
+	public function __construct(
+		$appName,
+		IRequest $request,
+		IConfig $config,
+		IUserManager $userManager,
+		IUserSession $userSession,
+		IRootFolder $rootFolder
+	) {
+		parent::__construct($appName, $request);
+		$this->config = $config;
+		$this->userSession = $userSession;
+		$this->root = $rootFolder;
+	}
+
+    /**
+     * @NoAdminRequired
+     */
+    public function setNotesPath($notesPath) {
+	    $uid = $this->userSession->getUser()->getUID();
+
+	    $path = '/' . $uid . '/files' . $notesPath;
```

Comment:
I think we should sanitize `$notesPath` before using it here. However, I wasn't able to attack this. Does the Files API checks for directory traversals?

Reply:
I can't find anything about this in the dev docs. The closest info I've found is this: https://github.com/nextcloud/server/blob/stable13/lib/public/Files/Folder.php#L104
Since the path was not "manually" sanitized before my changes, I did not sanitize them, too and I assume that this is done internally. But you're right, this should be verified - but I think I'm not the right person to do so :)